test/libyul/yulOptimizerTests/fullSuite/ackermann_function.yul

```diff
@@ -0,0 +1,559 @@
+// https://en.wikipedia.org/wiki/Ackermann_function
+// Test to see how FunctionSpecializer deals with functions that are too recursive / resource intensive.
+{
+    // 5
+    sstore(0, A(2, 1))
+    // 7
+    sstore(1, A(2, 2))
+
+    // Too many unrolling needed. In arbitrary precision, the value is 2**65536 - 3.
+    sstore(2, A(4, 2))
+
+    function A(m, n) -> ret {
+        switch m
+        case 0 { ret := add(n, 1) }
+        default {
+            switch n
+            case 0 { ret := A(sub(m, 1), 1) }
+            default { ret := A(sub(m, 1), A(m, sub(n, 1))) }
+        }
+    }
+}
+// ----
+// step: fullSuite
+//
+// {
+//     {
+//         sstore(0, 5)
+//         sstore(1, add(2, 5))
+//         let _1 := not(0)
+//         let _2 := A_985(A_4878(A_4886(add(5, _1))))
+//         let ret := 0
+//         switch _2
+//         case 0 { ret := 5 }
+//         default {
+//             let _3 := A_985(add(_2, _1))
+//             let ret_1 := 0
+//             switch _3
+//             case 0 { ret_1 := 3 }
+//             default {
+//                 ret_1 := A_2511(A_2505(add(_3, _1)))
+//             }
+//             ret := ret_1
+//         }
+//         sstore(2, ret)
+//     }
+//     function A_294_20862() -> ret
+//     {
+//         switch 1
+//         case 0 { ret := add(ret, 1) }
```

Comment:
Interesting - why doesn't it use `ret = 0`?

Reply:
I think we reached the `maxRounds` for `runSequenceUntilStable`.